src/wallet/did_wallet/did_wallet_puzzles.py

```diff
@@ -0,0 +1,111 @@
+from src.types.sized_bytes import bytes32
+from src.types.program import Program
+from typing import List, Optional, Tuple
+from blspy import G1Element
+from src.types.coin import Coin
+from src.types.coin_solution import CoinSolution
+from src.util.ints import uint64
+from src.wallet.puzzles.load_clvm import load_clvm
+import clvm
+
+DID_CORE_MOD = load_clvm("did_core.clvm")
+DID_INNERPUZ_MOD = load_clvm("did_innerpuz.clvm")
+DID_RECOVERY_MESSAGE_MOD = load_clvm("did_recovery_message.clvm")
+DID_GROUP_MOD = load_clvm("did_groups.clvm")
+
+
+def create_innerpuz(pubkey: bytes, identities: List[bytes]) -> Program:
+    return DID_INNERPUZ_MOD.curry(DID_CORE_MOD.get_tree_hash(), pubkey, identities)
+
+
+def create_fullpuz(innerpuz, genesis_id) -> Program:
+    mod_hash = DID_CORE_MOD.get_tree_hash()
+    return DID_CORE_MOD.curry(mod_hash, genesis_id, innerpuz)
+
+
+def fullpuz_hash_for_inner_puzzle_hash(
+    mod_code, genesis_id, inner_puzzle_hash
+) -> bytes32:
+    """
+    Given an inner puzzle hash, calculate a puzzle program hash for a specific cc.
+    """
+    gid_hash = genesis_id.get_tree_hash()
+    return mod_code.curry(
+        mod_code.get_tree_hash(), gid_hash, inner_puzzle_hash
+    ).get_tree_hash(gid_hash, inner_puzzle_hash)
+
+
+def get_pubkey_from_innerpuz(innerpuz: Program) -> G1Element:
+    pubkey_program, id_list = uncurry_innerpuz(innerpuz)
+    pubkey = G1Element.from_bytes(pubkey_program.as_atom())
+    return pubkey
+
+
+def is_did_innerpuz(inner_f: Program):
```

Comment:
Why do you need this function? If the inner puzzle can only ever be one thing, why not just include it in the main `MOD`? And if it can't only be one thing, this check is invalid.